index.js

```diff
@@ -199,7 +199,11 @@ function hash(src, filePath) {
 }
 
 function symlinkOrCopyFromCache(entry, dest, relativePath) {
-  mkdirp.sync(path.dirname(entry.outputFile));
-
-  symlinkOrCopySync(entry.cacheFile, dest + '/' + relativePath);
+  try {
```

Comment:
Ya, we definitely could if you'd prefer that approach. We should co-opt that for broccoli-persistent-filter too though (I took inspiration here from that).